ReactAndroid/src/main/java/com/facebook/react/ReactRootView.java

```diff
@@ -404,13 +408,37 @@ private void checkForDeviceOrientationChanges() {
         return;
       }
       mDeviceRotation = rotation;
-      // It's important to repopulate DisplayMetrics and export them before emitting the
-      // orientation change event, so that the Dimensions object returns the correct new values.
-      DisplayMetricsHolder.initDisplayMetrics(getContext());
-      emitUpdateDimensionsEvent();
       emitOrientationChanged(rotation);
     }
 
+    private void checkForDeviceDimensionsChanges() {
+      // Get current display metrics.
+      DisplayMetricsHolder.initDisplayMetrics(getContext());
+      // Check changes to both window and screen display metrics since they may not update at the same time.
+      if (!areMetricsEqual(mWindowMetrics, DisplayMetricsHolder.getWindowDisplayMetrics())
+        || !areMetricsEqual(mScreenMetrics, DisplayMetricsHolder.getScreenDisplayMetrics())) {
+        mWindowMetrics.setTo(DisplayMetricsHolder.getWindowDisplayMetrics());
+        mScreenMetrics.setTo(DisplayMetricsHolder.getScreenDisplayMetrics());
+        emitUpdateDimensionsEvent();
+      }
+    }
+
+    private boolean areMetricsEqual(DisplayMetrics displayMetrics, DisplayMetrics otherMetrics) {
+      if (Build.VERSION.SDK_INT >= 17) {
+        return displayMetrics.equals(otherMetrics);
+      } else {
+        // DisplayMetrics didn't have an equals method before API 17.
+        // Check all public fields manually.
+        return displayMetrics.widthPixels == otherMetrics.widthPixels
+          && displayMetrics.heightPixels == otherMetrics.heightPixels
```

Comment:
I see. Yes that seems to be the pattern. I'll fix that.

Indentation for the if condition did not match the current style either as per line [147](https://github.com/atsjenk/react-native/blob/e99afe23e45d8140dc0c827f833a862d3ae270f7/ReactAndroid/src/main/java/com/facebook/react/ReactRootView.java#L147).